vms/avm/wallet_service.go

```diff
@@ -31,27 +31,26 @@ type WalletService struct {
 }
 
 func (w *WalletService) decided(txID ids.ID) {
-	if _, ok := w.pendingTxs.Get(txID); !ok {
+	if !w.pendingTxs.Delete(txID) {
 		return
 	}
 
 	w.vm.ctx.Log.Info("tx decided over wallet API",
 		zap.Stringer("txID", txID),
 	)
-	w.pendingTxs.Delete(txID)
-
 	for {
 		txID, tx, ok := w.pendingTxs.Oldest()
 		if !ok {
 			return
 		}
 
-		txBytes := tx.Bytes()
-		_, err := w.vm.IssueTx(txBytes)
+		err := w.vm.mempool.Add(tx)
 		if err == nil {
 			w.vm.ctx.Log.Info("issued tx to mempool over wallet API",
 				zap.Stringer("txID", txID),
 			)
+
```

Comment:
afaict, it wouldn't be easy to change this to be called after all txs in a block have `onAccept`/`onReject` on them